ui/app/AppLayouts/Wallet/views/CollectiblesView.qml

```diff
@@ -53,8 +56,68 @@ ColumnLayout {
 
         readonly property bool isCustomView: cmbTokenOrder.currentValue === SortOrderComboBox.TokenOrderCustom
 
-        readonly property bool hasCollectibles: nonCommunityModel.count
-        readonly property bool hasCommunityCollectibles: communityModel.count
+        readonly property var sourceModel: root.controller.sourceModel
+        readonly property bool isLoading: root.isUpdating || root.isFetching
+
+        onIsLoadingChanged: {
+            d.loadingItemsModel.refresh()
+        }
+
+        readonly property var loadingItemsModel: ListModel {
+            Component.onCompleted: {
+                refresh()
+            }
+
+            function refresh() {
+                clear()
+                if (d.isLoading) {
+                    for (let i = 0; i < 10; i++) {
+                        append({ isLoading: true })
+                    }
+                }
+            }
+        }
+
+        readonly property var communityModel: CustomSFPM {
+            isCommunity: true
+        }
+
+        readonly property var communityModelWithLoadingItems: ConcatModel {
+            sources: [
+                SourceModel {
+                    model: d.communityModel
+                    markerRoleValue: "communityModel"
+                },
+                SourceModel {
+                    model: d.loadingItemsModel
+                    markerRoleValue: "loadingItemsModel"
+                }
+            ]
+
+            markerRoleName: "sourceGroup"
+        }
+
+        readonly property var nonCommunityModel: CustomSFPM {
+            isCommunity: false
+        }
+
+        readonly property var nonCommunityModelWithLoadingItems: ConcatModel {
+            sources: [
+                SourceModel {
+                    model: d.nonCommunityModel
+                    markerRoleValue: "nonCommunityModel"
+                },
+                SourceModel {
+                    model: d.loadingItemsModel
+                    markerRoleValue: "loadingItemsModel"
+                }
+            ]
+
+            markerRoleName: "sourceGroup"
+        }
+
+        readonly property bool hasCollectibles: d.nonCommunityModelWithLoadingItems.rowCount
```

Comment:
Hmm the View's count changes to 0 when collapsing the category, so I cannot use that.
How about exposing a count property in ConcatModel? Otherwise I can just manually check the counts of the separate source models.

Reply:
For now, doing the latter for simplicity. Can create a followup issue if the added `count` property were desired.